sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -817,6 +823,25 @@ class SessionCatalog(
     }
   }
 
+  def parseTempViewPlan(plan: LogicalPlan): LogicalPlan = {
+    plan match {
+      case viewInfo: TemporaryViewRelation =>
+        val metadata = viewInfo.tableMeta
```

Comment:
to remove duplicated code, we can have a function that turns `CatalogTable` to `View`, and takes a `isTemp` flag.